File: solutions/java/word-count/2/src/main/java/WordCount.java
import java.util.HashMap;
import java.util.Map;

class WordCount {
    public Map<String, Integer> phrase(String input) {

        Map<String, Integer> result = new HashMap<>();
        input = input.toLowerCase().replaceAll("[^a-z1-9']|\\B'|'\\B", " ").trim();
        var words = input.split("\\s+");

        for (String word : words) {
            result.put(word, result.get(word) == null ? 1 : result.get(word) + 1);
        }

        return result;
    }
}

File: solutions/go/word-count/2/word_count.go
package wordcount

import (
	"regexp"
	"strings"
)

type Frequency map[string]int

var reg = regexp.MustCompile(`[^a-zA-Z0-9']+`)

func WordCount(phrase string) Frequency {
	words := strings.Fields(normalize(phrase))
	freq := Frequency{}
	for _, word := range words {
		word = strings.Trim(word, "'")
		freq[word]++
	}
	return freq
}

func normalize(phrase string) string {
	phrase = strings.ToLower(phrase)
	return reg.ReplaceAllString(phrase, " ")
}
